app/connectors/AnswersConnector.scala

```diff
@@ -39,15 +39,11 @@ class AnswersConnector @Inject()(val repository: AnswersRepository)(implicit ec:
       case None                => save(UserAnswers(eori))
     }
 
-  def upsert(answers: UserAnswers): Future[UserAnswers] = {
+  def upsert(answers: UserAnswers): Future[Option[UserAnswers]] = {
```

Comment:
Should change this name as it is no longer 'upserting', its just a normal update now that might not find any matches to update

Reply:
Actually, there is an error. It SHOULD upsert if document is not found.